client/my-sites/drafts/draft-list.jsx

```diff
@@ -1,62 +1,93 @@
-	observe = require( 'lib/mixins/data-observe' );
-
-var DraftList = React.createClass( {
-
-	mixins: [ PureRenderMixin ],
+import AuthorSelector from './author-selector';
+import Count from 'components/count';
+import Draft from 'my-sites/draft';
+import EmptyContent from 'components/empty-content';
+import PostListFetcher from 'components/post-list-fetcher';
+import QueryPostCounts from 'components/data/query-post-counts';
+import {
+	getNormalizedPostCounts,
+	getNormalizedMyPostCounts,
+	isRequestingPostCounts
+} from 'state/posts/counts/selectors';
+import actions from 'lib/posts/actions';
+import { hasTouch } from 'lib/touch-detect';
+import infiniteScroll from 'lib/mixins/infinite-scroll';
+import observe from 'lib/mixins/data-observe';
+import userLib from 'lib/user';
+const user = userLib();
+
+class DraftList extends Component {
+
+	static propTypes = {
+		search: PropTypes.string,
+		sites: PropTypes.object,
+		siteID: PropTypes.any,
+		trackScrollPage: PropTypes.func,
+		onTitleClick: PropTypes.func,
+		showAllActionsMenu: PropTypes.bool,
+		selectedId: PropTypes.number
+	};
+
+	static defaultProps = {
+		showAllActionsMenu: true,
+		selectedId: false
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			authorFilter: 'me'
+		};
+	}
 
-	propTypes: {
-		search: React.PropTypes.string,
-		sites: React.PropTypes.object,
-		siteID: React.PropTypes.any,
-		trackScrollPage: React.PropTypes.func,
-		onTitleClick: React.PropTypes.func,
-		showAllActionsMenu: React.PropTypes.bool,
-		selectedId: React.PropTypes.number
-	},
+	setAuthorFilter = authorFilter => {
+		this.setState( { authorFilter } );
+	}
 
-	getDefaultProps: function() {
-		return {
-			showAllActionsMenu: true,
-			selectedId: false
-		}
-	},
+	render() {
+		const { siteID, counts, isRequestingCounts, selectedId } = this.props;
+		const { authorFilter } = this.state;
+		const author = authorFilter === 'me' ? user.get().ID : null;
+		const showCount = siteID && ( ! isRequestingCounts || counts[ authorFilter ] );
 
-	render: function() {
 		return (
-			<PostListFetcher
-				siteID={ this.props.siteID }
-				status="draft,pending"
-				withImages={ true }
-				withCounts={ true }
-				onTitleClick={ this.props.onTitleClick }
-			>
-				<Drafts
-					{ ...omit( this.props, 'children' ) }
-					status="draft"
-					selectedId={ this.props.selectedId }
-				/>
-			</PostListFetcher>
+			<div>
+				{ siteID && <QueryPostCounts siteId={ siteID } type="post" /> }
+				<div className="drafts__header">
+					<AuthorSelector selectedScope={ authorFilter }
+						onChange={ this.setAuthorFilter } />
+					{ showCount && <Count count={ counts[ authorFilter ] } /> }
+				</div>
+				<PostListFetcher
+					siteID={ this.props.siteID }
+					status="draft,pending"
+					author={ author }
+					withImages={ true }
+					withCounts={ true }
+					onTitleClick={ this.props.onTitleClick }
+				>
+					<Drafts
+						{ ...omit( this.props, 'children' ) }
```

Comment:
`<Drafts />` doesn't do anything with `children` anyways, so why do we omit it? Wasted time.